node_modules/oae-principals/tests/test-download-log.js

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2018 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var PrincipalsTestUtil = require('oae-principals/lib/test/util');
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests');
+
+var PrincipalsDAO = require('oae-principals/lib/internal/dao');
+var PrincipalsTestUtil = require('oae-principals/lib/test/util');
+
+describe('Download Logs', function() {
+
+    // Rest contexts that will be used for requests
+    var anonymousRestContext = null;
+    var camAdminRestContext = null;
+    var gtAdminRestContext = null;
+
+
+    /**
+     * Function that will fill up the anonymous and tenant admin REST context
+     */
+    before(function(callback) {
+        anonymousRestContext = TestsUtil.createTenantRestContext(global.oaeTests.tenants.cam.host);
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+        gtAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.gt.host);
+        return callback();
+    });
+
+    var parseUser = function(hash) {
+        var user = {
+            'principalId': hash.user.id,
+            'tenantAlias': hash.user.tenant.alias,
+            'email': hash.user.email,
+            'publicAlias': hash.user.displayName,
+            'created': hash.user.created,
+            'lastLog': hash.user.lastLog
+        };
+        return user;
+    };
+
+    var sortList = function(list) {
+        return list.sort(function (a, b) {
+            if (a < b) return -1;
+            else if (a > b) return 1;
+            return 0;
+        });
+    };
+
+    describe('Download the log of the user creation date', function() {
+
+        /**
+         * Test that verifies that we get the right users
+         */
+        it('verify if we get the right users on download log of user creation date', function(callback) {
+            // Create user
+            TestsUtil.generateTestUsers(camAdminRestContext, 2, function(err, users, mrvisser, simong) {
+                assert.ok(!err);
+
+                camAdminRestContext.user = function () { return mrvisser.user; };
+                camAdminRestContext.tenant = function () { return mrvisser.user.tenant; };
+                camAdminRestContext.user().isAdmin = function () { return true; };
```

Comment:
I need these informations because in the function located in api.user.js, I check if the user is an admin or not. I also need the tenantAlias. 
Maybe there is a cleaner way to do that ^^'

Reply:
You're right, I didn't realise that. 👍